Bottom: 04af7949bb94b9d54e62f411da0231457fb23ca7
Top:    3934c2ece50e5dc74a24fa77abbba27ece39a7cb
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2015-01-04 01:50:18 +0000

Fix to ParseGroupString()


---

diff --git a/src/misc.c b/src/misc.c
index 31021c9..b01a5ae 100644
--- a/src/misc.c
+++ b/src/misc.c
@@ -1101,7 +1101,14 @@ ParseGroupString (char *group_string, LayerGroupType *LayerGroup, int *LayerN)
           /* ignore white spaces and check for separator */
           while (*s && isspace ((int) *s))
             s++;
-          if (!*s || *s == ':')
+
+          /* Backup so the outer loop terminates on the NULL termination */
+          if (*s == '\0') {
+            s--;
+            break;
+          }
+
+          if (*s == ':')
             break;
         }
       LayerGroup->Number[group] = member;
